pkg/controllers/provisioning/provisioner.go

```diff
@@ -111,7 +111,8 @@ func (p *Provisioner) Builder(_ context.Context, mgr manager.Manager) operatorco
 			// wait to ensure that our cluster state is synced with the current known nodes to prevent over-provisioning
 			for WaitForClusterSync {
 				if err := p.cluster.Synchronized(ctx); err != nil {
-					logging.FromContext(ctx).Infof("waiting for cluster state to catch up, %s", err)
+					ctx = logging.WithLogger(ctx, logging.FromContext(ctx).With("clusterState", err))
```

Comment:
```suggestion
```

Errors don't really work well as structured values since they aren't consistent and hard to parse